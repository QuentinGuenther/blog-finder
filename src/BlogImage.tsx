import React, {useState} from 'react';

export const DEFAULT_IMG_URL = 'default image path';

export interface BlogImageProps {
  imgPath?: string,
  altText: string
}

export function BlogImage (props: BlogImageProps) {
  const src = (props.imgPath) ? props.imgPath : DEFAULT_IMG_URL;

  return (
    <img src={src} alt={props.altText}/>
  );
}
