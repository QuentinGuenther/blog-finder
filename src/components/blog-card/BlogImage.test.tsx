import * as React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { BlogImage, DEFAULT_IMG_URL } from './BlogImage';

describe('BlogImage should match snapshot', () => {
  it('Interaction should match snapshot', () => {
    const tree = renderer
      .create(<BlogImage imgPath='./test.png' altText='test' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('BlogImage should handle working images', () => {
  it('Working image path should display', () => {
    const blogImage = shallow(<BlogImage imgPath='./index.png' altText='this is alt' />);
    expect(blogImage.find('img').at(0).props().src).toEqual('./index.png');
    expect(blogImage.find('img').at(0).props().alt).toEqual('this is alt');
  });
});

describe('BlogImage should handle broken images', () => {
  it('Missing image path should default image src', () => {
    const blogImage = shallow(<BlogImage altText='this is default' />);
    expect(blogImage.find('img').at(0).props().src).toEqual(DEFAULT_IMG_URL);
    expect(blogImage.find('img').at(0).props().alt).toEqual('this is default');
  });
});