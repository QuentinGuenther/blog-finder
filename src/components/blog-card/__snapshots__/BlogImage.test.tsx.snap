// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogImage should match snapshot Interaction should match snapshot 1`] = `
<img
  alt="test"
  src="./test.png"
/>
`;
